.circleci/config.yml

```diff
@@ -1,23 +1,40 @@
-version: 2
-jobs:
+version: 2.1
+
+jobs: 
   build:
     docker:
-      - image: circleci/node:10
+      - image: circleci/node
+
+    parameters:
+      eslint:
+        type: string
 
     steps:
       - checkout
 
       - restore_cache:
           keys:
-          - v1-dependencies-{{ checksum "yarn.lock" }}
+          - v1-dependencies-{{ checksum "yarn.lock" }}-<< parameters.eslint >>
           - v1-dependencies-
-
+      
       - run: yarn install --frozen-lock
 
       - save_cache:
           paths:
             - node_modules
-          key: v1-dependencies-{{ checksum "yarn.lock" }}
-      
-      - run: yarn format -- --list-different
+          key: v1-dependencies-{{ checksum "yarn.lock" }}-<< parameters.eslint >>
+
+      - run: yarn add --dev eslint@<< parameters.eslint >>
+
+      - run: yarn format:check
+      - run: yarn lint
       - run: yarn test
```

Comment:
Just to confirm, from the compatibility perspective, we're more interested in the test results because `yarn lint` just lints the source in this repo?

Reply:
Yes, you are right.

Reply:
Yes, you are right.